resources/qml/UserProfile.qml

```diff
@@ -34,279 +34,287 @@ ApplicationWindow {
-                anchors.leftMargin: Nheko.paddingSmall
-                anchors.left: displayUsername.right
-                anchors.verticalCenter: displayUsername.verticalCenter
-                hoverEnabled: true
-                ToolTip.visible: hovered
-                ToolTip.text: profile.isGlobalUserProfile ? qsTr("Change display name globally.") : qsTr("Change display name. Will only apply to this room.")
-                image: displayUsername.isUsernameEditingAllowed ? ":/icons/icons/ui/checkmark.png" : ":/icons/icons/ui/edit.png"
-                onClicked: {
-                    if (displayUsername.isUsernameEditingAllowed) {
-                        profile.changeUsername(displayUsername.text);
-                        displayUsername.isUsernameEditingAllowed = false;
-                    } else {
-                        displayUsername.isUsernameEditingAllowed = true;
-                        displayUsername.focus = true;
-                        displayUsername.selectAll();
+            TextInput {
+                id: displayUsername
+
+                property bool isUsernameEditingAllowed
+
+                readOnly: !isUsernameEditingAllowed
+                text: profile.displayName
+                font.pixelSize: 20
+                color: TimelineManager.userColor(profile.userid, Nheko.colors.window)
+                font.bold: true
+                Layout.alignment: Qt.AlignHCenter
+                selectByMouse: true
+                onAccepted: {
+                    profile.changeUsername(displayUsername.text);
+                    displayUsername.isUsernameEditingAllowed = false;
+                }
+
+                ImageButton {
+                    visible: profile.isSelf
+                    anchors.leftMargin: Nheko.paddingSmall
+                    anchors.left: displayUsername.right
+                    anchors.verticalCenter: displayUsername.verticalCenter
+                    hoverEnabled: true
+                    ToolTip.visible: hovered
+                    ToolTip.text: profile.isGlobalUserProfile ? qsTr("Change display name globally.") : qsTr("Change display name. Will only apply to this room.")
+                    image: displayUsername.isUsernameEditingAllowed ? ":/icons/icons/ui/checkmark.png" : ":/icons/icons/ui/edit.png"
+                    onClicked: {
+                        if (displayUsername.isUsernameEditingAllowed) {
+                            profile.changeUsername(displayUsername.text);
+                            displayUsername.isUsernameEditingAllowed = false;
+                        } else {
+                            displayUsername.isUsernameEditingAllowed = true;
+                            displayUsername.focus = true;
+                            displayUsername.selectAll();
+                        }
                     }
                 }
+
             }
 
-        }
+            MatrixText {
+                text: profile.userid
+                font.pixelSize: 15
```

Comment:
We should probably fix that to be a multiple of the configured font size. I know, not your fault, so you don't really have to fix it.